processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -146,7 +147,9 @@ public SegmentMetadataQueryRunnerFactory(
                           @Override
                           public Sequence<SegmentAnalysis> call() throws Exception
                           {
-                            return input.run(query, responseContext);
```

Comment:
@nishantmonu51  that might be a good idea, I will see if we can use ChainedExecutionQueryRunner here as well.